### ISSUE TITLE
Add crypto as an option

### ISSUE THREAD

**Issue by genonullfree** [genonullfree/teleport#51]
Bring back ECDH crypto! (optionally)
Add flag `-e` option to exchange elliptic curve public keys and encrypt the file transfer.
The `ecdh` branch may yet be of some use to us.

**Comment by genonullfree** [comment]
testing current version (`v0.5.3`) against the `ecdh` shows `ecdh` being almost exactly 10x slower than `v0.5.3`...
```
# v0.5.3
cargo run -- -i ~/Downloads/Wireshark\ 3.4.9\ Intel\ 64.dmg  2.84s user 0.28s system 99% cpu 3.128 total
# ecdh
cargo run -- -i ~/Downloads/Wireshark\ 3.4.9\ Intel\ 64.dmg  28.91s user 0.25s system 96% cpu 30.132 total
```

**Comment by genonullfree** [comment]
This is currently being worked on in the new v0.6.0 protocol spec implementation, which will be quite an overhaul from the <= v0.5.x series.

**Comment by genonullfree** [comment]
Implemented `v0.6.0` with crypto, here are some stats:

**No encryption (standard)**
```
geno@foundation teleport % cargo run -- -i ~/Downloads/ubuntu-20.04.3-live-server-arm64.iso -kon 
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/teleporter -i /Users/geno/Downloads/ubuntu-20.04.3-live-server-arm64.iso -kon`
Teleporter Client 0.6.0
Sending file 1/1: /Users/geno/Downloads/ubuntu-20.04.3-live-server-arm64.iso
 =>    1.145G of    1.145G (100.00%) done! Time: 24.95s Speed: 390.689 Mbps
```
**Encryption (`-e`)**
```
geno@foundation teleport % cargo run -- -i ~/Downloads/ubuntu-20.04.3-live-server-arm64.iso -kone
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/teleporter -i /Users/geno/Downloads/ubuntu-20.04.3-live-server-arm64.iso -kone`
Teleporter Client 0.6.0
Sending file 1/1: /Users/geno/Downloads/ubuntu-20.04.3-live-server-arm64.iso
 =>    1.145G of    1.145G (100.00%) done! Time: 319.52s Speed: 29.394 Mbps
```

Previous version (`v0.5.5`) for speed reference:
```
geno@foundation teleport % cargo run -- -i ~/Downloads/ubuntu-20.04.3-live-server-arm64.iso      
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/teleporter -i /Users/geno/Downloads/ubuntu-20.04.3-live-server-arm64.iso`
Teleporter Client 0.5.5
Sending file 1/1: "ubuntu-20.04.3-live-server-arm64.iso"
 =>    1.145G of    1.145G (100.00%) done! Time: 28.59s Speed: 334.877 Mbps
```
